aiokafka/producer/message_accumulator.py

```diff
@@ -165,7 +166,9 @@ def failure(self, exception):
         for future, _ in self._msg_futures:
             if future.done():
                 continue
-            future.set_exception(exception)
+            # we need to copy exception so traceback is not multiplied
+            # https://github.com/aio-libs/aiokafka/issues/246
+            future.set_exception(copy.deepcopy(exception))
```

Comment:
Seems like copy.copy() is enough to prevent traceback changes. The tracebacks are immutable and linked in a linked list, so copy is enough.
For example:
```python
import copy


def test_raise_many():
    err = ValueError()
    for i in range(100):
        try:
            raise copy.copy(err)
        except Exception:
            pass
    raise err

test_raise_many()
```

Reply:
See https://gitter.im/aio-libs/Lobby?at=5a0d094871ad3f8736edbd5b reply on gitter also